benchmarks/benchmarks/sparse_linalg_lobpcg.py

```diff
@@ -66,45 +29,119 @@ def __init__(self):
         self.time_mikota.__func__.setup = self.setup_mikota
 
         self.time_sakurai.__func__.params = list(self.params)
-        self.time_sakurai.__func__.params[0] = [50, 400]
+        self.time_sakurai.__func__.params[0] = [50]
         self.time_sakurai.__func__.setup = self.setup_sakurai
 
+        self.time_sakurai_inverse.__func__.params = list(self.params)
+        self.time_sakurai_inverse.__func__.params[0] = [500, 1000]
+        self.time_sakurai_inverse.__func__.setup = self.setup_sakuraii
+
+
     def setup_mikota(self, n, solver):
         self.shape = (n, n)
-        self.A, self.B = _mikota_pair(n)
-
-        if solver == 'eigh' and n >= 512:
+        mik = MikotaPair(n)
+        mik_k = mik.k
+        mik_m = mik.m
+        self.Ac = mik_k
+        self.Aa = mik_k.toarray()
+        self.Bc = mik_m
+        self.Ba = mik_m.toarray()
+        self.Ab = mik_k.tobanded()
+        self.eigenvalues = mik.eigenvalues
+
+        if solver == 'lapack' and n > 512:
             # skip: slow, and not useful to benchmark
             raise NotImplementedError()
 
     def setup_sakurai(self, n, solver):
         self.shape = (n, n)
-        self.A, self.B, all_eigenvalues = _sakurai(n)
-        self.A_dense = self.A.A
-        self.B_dense = self.B.A
+        sakurai_obj = Sakurai(n, dtype='int')
+        self.A = sakurai_obj
+        self.Aa = sakurai_obj.toarray()
+        self.eigenvalues = sakurai_obj.eigenvalues
 
+    def setup_sakuraii(self, n, solver):
```

Comment:
I think you have to match the `setup_` naming with the `test_` naming
```suggestion
    def setup_sakurai_inverse(self, n, solver):
```